src/osha/oira/statistics/utils.py

```diff
@@ -202,20 +256,32 @@ def account_rows(offset):
                 )
                 for account in batch
             ]
-            return rows
+            self.session_statistics.add_all(rows)
+            return len(rows)
 
-        log.info("Table: account")
         self._process_batch(account_rows)
 
     def update_company(self, country=None):
-        companies = (
-            self.session_application.query(Company, SurveySession.zodb_path)
-            .filter(Company.session_id == SurveySession.id)
-            .order_by(Company.id)
+        log.info("Table: company")
+        latest_date = (
+            self.session_statistics.query(
+                sqlalchemy.func.max(CompanyStatistics.date)
+            ).first()[0]
+            or datetime.min
         )
+        companies = self.session_application.query(Company, SurveySession.zodb_path)
+
+        if latest_date > datetime.min:
+            log.info(f"Skipping company responses up to and including {latest_date}")
+            companies = companies.filter(Company.timestamp > latest_date)
+
```

Comment:
The same comments I did before about the use of first and the f-string also apply here